lib/std/zig/Ast.zig

```diff
@@ -101,6 +116,13 @@ pub fn parse(gpa: Allocator, source: [:0]const u8, mode: Mode) Allocator.Error!A
     }
 
     // TODO experiment with compacting the MultiArrayList slices here
+
+    // Save memory by freeing the allocated (but not used) sections:
+    tokens.shrinkAndFree(gpa, tokens.len);
+    parser.nodes.shrinkAndFree(gpa, parser.nodes.len);
+    parser.extra_data.shrinkAndFree(gpa, parser.extra_data.items.len);
+    parser.errors.shrinkAndFree(gpa, parser.errors.items.len);
```

Comment:
I believe the CI failures are caused by these `shrinkAndFree` calls.

- The failures are from `error.SwallowedOutOfMemoryError` during `checkAllAllocationFailures` in the parser tests [here](https://github.com/ziglang/zig/blob/master/lib/std/zig/parser_test.zig)
- The `SwallowedOutOfMemoryError` is because `shrinkAndFree` does indeed 'swallow' `error.OutOfMemory` [here](https://github.com/ziglang/zig/blob/218cf059dd215282aa96d6b4715e68d533a4238e/lib/std/array_list.zig#L1042-L1048)

From the [doc comment of `checkAllAllocationFailures`](https://github.com/ziglang/zig/blob/218cf059dd215282aa96d6b4715e68d533a4238e/lib/std/testing.zig#L963-L1038):

```
/// Expects that `test_fn` has a deterministic number of memory allocations:
/// - If an allocation was made to fail during a run of `test_fn`, but `test_fn`
///   didn't return `error.OutOfMemory`, then `error.SwallowedOutOfMemoryError`
///   is returned from `checkAllAllocationFailures`. You may want to ignore this
///   depending on whether or not the code you're testing includes some strategies
///   for recovering from `error.OutOfMemory`.
```

See [this blog post](https://www.ryanliptak.com/blog/zig-intro-to-check-all-allocation-failures/) for more details about `checkAllAllocationFailures`

The simplest fix would be to make a change to `testTransform` in `parser_test.zig` that looks something like:

```diff
-     return std.testing.checkAllAllocationFailures(fixed_allocator.allocator(), testTransformImpl, .{ &fixed_allocator, source, expected_source });
+     std.testing.checkAllAllocationFailures(fixed_allocator.allocator(), testTransformImpl, .{ &fixed_allocator, source, expected_source }) catch |err| switch (err) {
+         // ArrayList.shrinkAndFree swallows OutOfMemory
+         error.SwallowedOutOfMemoryError => return,
+         else => |e| return e,
+     };
```

Reply:
Thank you!